test/e2e/validation_test.go

```diff
@@ -532,20 +534,21 @@ func (tc *testContext) testInvalidServicesCM(cmName string) error {
 		return err
 	}
 	// Try to retrieve newly created ConfigMap and validate its contents
-	windowsServices, err := tc.waitForValidWindowsServicesConfigMap(cmName)
+	windowsServices, err := tc.waitForValidWindowsServicesConfigMap(cmName, expected)
 	if err != nil {
 		return errors.Wrapf(err, "error retrieving ConfigMap %s", cmName)
 	}
 	data, err := servicescm.Parse(windowsServices.Data)
-	if err != nil || data.ValidateRequiredContent() != nil {
+	if err != nil || data.ValidateExpectedContent(expected) != nil {
 		return errors.Wrapf(err, "error ensuring ConfigMap %s is re-created validly", cmName)
 	}
```

Comment:
Isn't this already done within `waitForValidWindowsServicesConfigMap()` which is called just above? We can remove this IMO (and update the error msg on line 539)